src/services/resource-service.ts

```diff
@@ -25,13 +25,24 @@ import {
   ApiMethodEnum,
   GetQuestion
 } from '~/types'
-import { createUrlPath } from '~/utils/helper-functions'
+import { createUrlPath, getFullUrl } from '~/utils/helper-functions'
+import { baseService } from './base-service'
 
 export const ResourceService = {
   getUsersLessons: async (
     params?: GetResourcesParams
   ): Promise<AxiosResponse<ItemsWithCount<Lesson>>> =>
     await axiosClient.get(URLs.resources.lessons.get, { params }),
+
+  getUsersLessonsQuery: (params?: GetResourcesParams) => {
+    return baseService.request<ItemsWithCount<Lesson>>({
+      method: 'GET',
+      url: getFullUrl({
+        pathname: URLs.resources.lessons.get,
+        searchParameters: params
```

Comment:
Are you sure that it's working without destructuring?
```suggestion
        searchParameters: { ...params }
```